File: mobile/components/Brand.js
import { View, Text, StyleSheet } from 'react-native';
import { useTheme, fonts, eyebrow } from '../theme';

// The rounded logo mark + serif wordmark used on splash and auth screens.
export function Logo({ size = 'md' }) {
  const { colors } = useTheme();
  const box = size === 'lg' ? 76 : 44;
  const radius = size === 'lg' ? 20 : 12;
  const glyph = size === 'lg' ? 34 : 20;
  const word = size === 'lg' ? 34 : 22;

  return (
    <View style={styles.row}>
      <View
        style={{
          width: box,
          height: box,
          borderRadius: radius,
          backgroundColor: colors.surfaceAlt,
          borderWidth: 1,
          borderColor: colors.border,
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Text style={{ fontSize: glyph, color: colors.primary }}>❋</Text>
      </View>
      <Text
        style={{
          fontFamily: fonts.display,
          fontSize: word,
          fontWeight: '700',
          color: colors.text,
          marginLeft: 12,
          letterSpacing: -0.4,
        }}
      >
        Noteriety
      </Text>
    </View>
  );
}

// The small monospace label with a leading rule, e.g. "—— WORKSPACE"
export function Eyebrow({ children, style }) {
  const { colors } = useTheme();
  return (
    <View style={[styles.eyebrowRow, style]}>
      <View style={[styles.rule, { backgroundColor: colors.primary }]} />
      <Text style={[eyebrow, { color: colors.primary }]}>{children}</Text>
    </View>
  );
}

// Serif page heading.
export function Display({ children, style, size = 30 }) {
  const { colors } = useTheme();
  return (
    <Text
      style={[
        {
          fontFamily: fonts.display,
          fontSize: size,
          fontWeight: '700',
          color: colors.text,
          letterSpacing: -0.6,
          lineHeight: size * 1.15,
        },
        style,
      ]}
    >
      {children}
    </Text>
  );
}

const styles = StyleSheet.create({
  row: { flexDirection: 'row', alignItems: 'center' },
  eyebrowRow: { flexDirection: 'row', alignItems: 'center', gap: 8 },
  rule: { width: 22, height: 2, borderRadius: 1 },
});

File: noteriety_mobile/lib/widgets/brand.dart
import 'package:flutter/material.dart';

import '../theme/app_colors.dart';
import '../theme/tokens.dart';

class Logo extends StatelessWidget {
  final String size; // 'md' | 'lg'
  final bool tile; // draw the rounded background square behind the logo
  const Logo({super.key, this.size = 'md', this.tile = false});

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    final isDark = context.isDark;
    final bool lg = size == 'lg';
    final double box = lg ? 76 : 44;
    final double radius = lg ? 20 : 12;
    final double art = (box * (tile ? 0.62 : 1)).roundToDouble();
    final double word = lg ? 34 : 22;

    return Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        Container(
          width: box,
          height: box,
          alignment: Alignment.center,
          decoration: BoxDecoration(
            color: tile ? colors.surfaceAlt : Colors.transparent,
            borderRadius: BorderRadius.circular(radius),
            border: tile ? Border.all(color: colors.border) : null,
          ),
          child: Image.asset(

            // import logos
            isDark
                ? 'assets/images/dark-theme.webp'
                : 'assets/images/light-theme.webp',
            width: art,
            height: art,
            fit: BoxFit.contain,
          ),
        ),
        const SizedBox(width: 12),
        Text(
          'Noteriety',
          style: TextStyle(
            fontFamilyFallback: AppFonts.serif,
            fontSize: word,
            fontWeight: FontWeight.w700,
            color: colors.text,
            letterSpacing: -0.4,
          ),
        ),
      ],
    );
  }
}

/// The small monospace label with a leading rule, e.g. "—— WORKSPACE".
class Eyebrow extends StatelessWidget {
  final String text;
  final EdgeInsetsGeometry? padding;
  const Eyebrow(this.text, {super.key, this.padding});

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Padding(
      padding: padding ?? EdgeInsets.zero,
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(
            width: 22,
            height: 2,
            margin: const EdgeInsets.only(right: 8),
            decoration: BoxDecoration(
              color: colors.primary,
              borderRadius: BorderRadius.circular(1),
            ),
          ),
          Text(
            text,
            style: TextStyle(
              fontFamilyFallback: AppFonts.mono,
              fontSize: 11,
              letterSpacing: 1.4,
              fontWeight: FontWeight.w600,
              color: colors.primary,
            ),
          ),
        ],
      ),
    );
  }
}

/// Serif page heading.
class Display extends StatelessWidget {
  final String text;
  final double size;
  final TextAlign? align;
  const Display(this.text, {super.key, this.size = 30, this.align});

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Text(
      text,
      textAlign: align,
      style: TextStyle(
        fontFamilyFallback: AppFonts.serif,
        fontSize: size,
        fontWeight: FontWeight.w700,
        color: colors.text,
        letterSpacing: -0.6,
        height: 1.15,
      ),
    );
  }
}